example-deployment/README.md

```diff
@@ -0,0 +1,329 @@
+```
+
+## Test the deployment
+
+If not done so already, launch the office demo
+
+```bash
+ros2 launch rmf_demos office.launch.xml headless:=true
+```
+
+Go to https://example.com/dashboard, if everything works, you should see a log in screen, use user=example, password=example.
+
+After that, you should be presented with the dashboard, you have successfully deployed `rmf-web`! 🎉
+
+# Troubleshooting
+
+## Deploying in an airgapped network
+
+There are certain parts that requires an internet connection to fetch the docker images and source codes, in order to deploy from within an airgapped network, you can use obtain the images first, then push them directly to minikube. For example
+
+when you have internet, run this to get the keycloak image
+```bash
+docker pull quay.io/keycloak/keycloak:12.0.4
+```
+
+now, connect to the airgapped network and push the image to minikube
+```bash
+.bin/minikube load quay.io/keycloak/keycloak:12.0.4
+```
+
+now you can deploy keycloak without require access to the internet
+
+If connection to the internet from the same PC is not possible, you can use `docker save` to save the image into a tarball, then transfer it to the minikube PC through whatever method possible (thumbdrives, cds etc) and use `docker load` to load the image. Then you can use `.bin/minikube load` to push it into minikube.
+
+
+## Updating the deployment
+
+If you make changes to the images, you will need to publish the new image and update the kubernetes resources.
+Since we are pushing the images directly to minikube, we can't tell kubernetes to pull the new image as our image doesn't actually exist in the registry. In order to tell kubernetes to use the new image, you need to delete the current deployment and re-deploy again.
+
+```bash
+.bin/minikube kubectl -- delete -f <path-to-k8s-file>
+.bin/minikube kubectl -- apply -f <path-to-k8s-file>
+```
+
+Note that when using the above method to delete the keycloak deployment, it will delete the persistent volume containing the database as well, to only delete the keycloak app, use
+
+```bash
+.bin/minikube kubectl -- delete -f k8s/keycloak.yaml -ltier=app
+```
+
+## Get a shell into the container
+
+It can be very useful to get a shell into the pod, kubernetes allow you to do this easily.
+
+```bash
+.bin/minikube kubectl -- get pods
+.bin/minikube kubectl -- exec -it <pod> -- bash
+```
+
+## I got a blank white page
+
+The most common cause is that connection to the keycloak server is not working, see if you can open the keycloak ui at https://example.com/auth.
+
+## It is stuck in "downloading building map..."
+
+This is usually because the ros2-bridge server can't connect to rmf, first make sure that
+
+* rmf_demos is running
+* your version of rmf_demos is exactly the same as the one used to build your ros2-bridge image.
+
+If it still doesn't work, get a shell into the ros2-bridge port
+
+```bash
+.bin/minikube kubectl -- exec -it deployment/ros2-bridge -- bash
```

Comment:
Somehow this command doesn't seem to work, but `.bin/minikube kubectl -- exec -it deployment/ros2-bridge bash` does, with a depreciation warning.

Reply:
https://github.com/kubernetes/minikube/issues/10791 it is a bug in minikube, I guess just leave it as it is for now as it is fixed in the next release, which should be released in a week or two.